sys/kern/imgact_elf.c

```diff
@@ -1724,8 +1724,8 @@ timekeep_cap(struct image_params *imgp)
 	KASSERT(timekeep_len == CHERI_REPRESENTABLE_LENGTH(timekeep_len),
 	    ("timekeep_len needs rounding"));
 
-	/* XXX: Read-only? */
-	return (cheri_capability_build_user_rwx(CHERI_CAP_USER_DATA_PERMS,
+	/* XXX: Should this be CHERI_CAP_USER_RODATA_PERMS? */
```

Comment:
I think you don't need LOADCAP for this one either do you?

I do think having an explicit "RODATA" constant would be good, but you would perhaps explicitly trim LOADCAP from this one as well.

Reply:
I've gone ahead and created CHERI_PERMS_USERSPACE_RODATA which does not include LOAD_CAP (I can maybe see a case for kernel-provided RO pages that do contain capabilities, but given a lack of I use case I've ignored the possibility.)